### PR TITLE
[Refactor] Reorder the method argument ordering in pkg/cmd

### LINKED ISSUE #1889: pkg/cmd: inconsistent arguments ordering
https://github.com/containerd/nerdctl/blob/35bb03ee856dd48c35171ff1c25d80d9239eb35b/pkg/cmd/apparmor/list_linux.go#L32

https://github.com/containerd/nerdctl/blob/35bb03ee856dd48c35171ff1c25d80d9239eb35b/pkg/cmd/builder/build.go#L45

https://github.com/containerd/nerdctl/blob/35bb03ee856dd48c35171ff1c25d80d9239eb35b/pkg/cmd/image/pull.go#L40

Probably `options` should be consistently placed in the last.

`stdout io.Writer` should be ~probably right after `ctx context.Context` (if any).~ put into the options struct, and the nil value should imply `os.Stdout`

### PR DESCRIPTION
**Pull request by Zheaoli**
Fix #1889 

Signed-off-by: Zheao.Li <me@manjusaka.me>

### CODE REVIEW

File: cmd/nerdctl/namespace_rm.go [hunk 1]
```diff
@@ -36,18 +36,21 @@ func newNamespaceRmCommand() *cobra.Command {
...
-		return types.NamespaceRemoveCommandOptions{}, err
+		return types.NamespaceRemoveOptions{}, err
 	}
-	return types.NamespaceRemoveCommandOptions{
+	return types.NamespaceRemoveOptions{
 		GOptions: globalOptions,
 		CGroup:   cgroup,
+		Stdout:   cmd.OutOrStdout(),
+		Stderr:   cmd.ErrOrStderr(),
+		Stdin:    cmd.InOrStdin(),
```
Comment (djdongjin) [inline cmd/nerdctl/namespace_rm.go:53]: `namespace.Remove` seems only use `cmd.OutOrStdout()`

File: cmd/nerdctl/network_rm.go [hunk 2]
```diff
@@ -44,10 +44,13 @@ func networkRmAction(cmd *cobra.Command, args []string) error {
...
 		return err
 	}
-	return network.Remove(cmd.Context(), types.NetworkRemoveCommandOptions{
+	return network.Remove(cmd.Context(), types.NetworkRemoveOptions{
 		GOptions: globalOptions,
 		Networks: args,
-	}, cmd.OutOrStdout())
+		Stdout:   cmd.OutOrStdout(),
+		Stderr:   cmd.ErrOrStderr(),
+		Stdin:    cmd.InOrStdin(),
```
Comment (djdongjin) [inline cmd/nerdctl/network_rm.go:52]: `network.Remove` seems only use `cmd.OutOrStdout()`

File: cmd/nerdctl/volume_inspect.go [hunk 3]
```diff
@@ -40,23 +40,24 @@ func newVolumeInspectCommand() *cobra.Command {
 	return volumeInspectCommand
 }
 
-func processVolumeInspectCommandOptions(cmd *cobra.Command) (types.VolumeInspectCommandOptions, error) {
+func processVolumeInspectCommandOptions(cmd *cobra.Command) (types.VolumeInspectOptions, error) {
```
Comment (djdongjin) [inline cmd/nerdctl/volume_inspect.go:43]: Do we also need to rename these `process${Command}CommandOptions` to `process${Command}Options`?

File: pkg/cmd/apparmor/list_linux.go [hunk 4]
```diff
@@ -29,14 +28,14 @@ import (
...
+func List(options types.ApparmorListOptions) error {
 	quiet := options.Quiet
-	w := stdout
+	w := options.Stdout
 	var tmpl *template.Template
 	format := options.Format
 	switch format {
 	case "", "table", "wide":
-		w = tabwriter.NewWriter(stdout, 4, 8, 4, ' ', 0)
+		w = tabwriter.NewWriter(options.Stdout, 4, 8, 4, ' ', 0)
```
Comment (djdongjin) [inline pkg/cmd/apparmor/list_linux.go:38]: Seems an existing duplication. we should either remove line 33 or use `w` here.

File: pkg/cmd/network/list.go [hunk 5]
```diff
@@ -38,16 +37,16 @@ type networkPrintable struct {
...
 	quiet := options.Quiet
 	format := options.Format
-	w := out
+	w := options.Stdout
 	var tmpl *template.Template
 
 	switch format {
 	case "", "table", "wide":
-		w = tabwriter.NewWriter(out, 4, 8, 4, ' ', 0)
+		w = tabwriter.NewWriter(options.Stdout, 4, 8, 4, ' ', 0)
```
Comment (djdongjin) [inline pkg/cmd/network/list.go:49]: Same here, we should remove line 44 or use `w` here

File: pkg/cmd/volume/list.go [hunk 6]
```diff
@@ -90,12 +89,12 @@ func removeSizeFilters(filters []string) []string {
...
 
-func lsPrintOutput(vols map[string]native.Volume, options types.VolumeListCommandOptions, stdout io.Writer) error {
-	w := stdout
+func lsPrintOutput(vols map[string]native.Volume, options types.VolumeListOptions) error {
+	w := options.Stdout
 	var tmpl *template.Template
 	switch options.Format {
 	case "", "table", "wide":
-		w = tabwriter.NewWriter(stdout, 4, 8, 4, ' ', 0)
+		w = tabwriter.NewWriter(options.Stdout, 4, 8, 4, ' ', 0)
```
Comment (djdongjin) [inline pkg/cmd/volume/list.go:97]: same here (line 93 and line 97)

File: pkg/composer/create.go [hunk 7]
```diff
@@ -43,7 +43,7 @@ const (
 	RecreateDiverged = "diverged"
 )
 
-// CreateOptions stores all option input from `nerdctl compose create`
+//  Options stores all option input from `nerdctl compose create`
```
Comment (djdongjin) [inline pkg/composer/create.go:46]: lint error here, change back

File: pkg/composer/exec.go [hunk 8]
```diff
@@ -25,7 +25,7 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// ExecOptions stores options passed from users as flags and args.
+// Options stores options passed from users as flags and args.
```
Comment (djdongjin) [inline pkg/composer/exec.go:28]: change back

File: pkg/composer/port.go [hunk 9]
```diff
@@ -24,7 +24,8 @@ import (
 	"github.com/containerd/nerdctl/pkg/containerutil"
 )
 
-// PortOptions has args for getting the public port of a given private port/protocol
+//	Options has args for getting the public port of a given private port/protocol
```
Comment (djdongjin) [inline pkg/composer/port.go:27]: change back

File: pkg/composer/restart.go [hunk 10]
```diff
@@ -28,7 +28,7 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// RestartOptions stores all option input from `nerdctl compose restart`
+// Options stores all option input from `nerdctl compose restart`
```
Comment (djdongjin) [inline pkg/composer/restart.go:31]: change back

File: pkg/composer/rm.go [hunk 11]
```diff
@@ -30,7 +30,8 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// RemoveOptions stores all options when removing compose containers:
+//	Options stores all options when removing compose containers:
```
Comment (djdongjin) [inline pkg/composer/rm.go:33]: change back

File: pkg/logging/json_logger.go [hunk 12]
```diff
@@ -137,7 +137,9 @@ func viewLogsJSONFile(lvopts LogViewOptions, stdout, stderr io.Writer, stopChann
 }
 
 // Loads JSON log entries directly from the provided JSON log file.
-// If `LogViewOptions.Follow` is provided, it will refresh and re-read the file until
+//
+//	# Options.Follow` is provided, it will refresh and re-read the file until
+//
```
Comment (djdongjin) [inline pkg/logging/json_logger.go:142]: change back

File: pkg/ipfs/registry.go [hunk 13]
```diff
@@ -39,7 +39,7 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// RegistryOptions represents options to configure the registry.
+// Options represents options to configure the registry.
```
Comment (djdongjin) [inline pkg/ipfs/registry.go:42]: change back

File: pkg/infoutil/infoutil_linux.go [hunk 14]
```diff
@@ -61,7 +61,8 @@ WARNING: AppArmor profile %q is not loaded.
 // fulfillPlatformInfo fulfills cgroup and kernel info.
 //
 // fulfillPlatformInfo requires the following fields to be set:
-// SecurityOptions, CgroupDriver, CgroupVersion
+//
+//	Options, CgroupDriver, CgroupVersion
```
Comment (djdongjin) [inline pkg/infoutil/infoutil_linux.go:65]: change back

File: pkg/imgutil/dockerconfigresolver/dockerconfigresolver.go [hunk 15]
```diff
@@ -88,7 +88,7 @@ func WithAuthCreds(ac AuthCreds) Opt {
 	}
 }
 
-// NewHostOptions instantiates a HostOptions struct using $DOCKER_CONFIG/config.json .
+//	Options instantiates a HostOptions struct using $DOCKER_CONFIG/config.json .
```
Comment (djdongjin) [inline pkg/imgutil/dockerconfigresolver/dockerconfigresolver.go:91]: change back

File: pkg/composer/stop.go [hunk 16]
```diff
@@ -29,7 +29,7 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// StopOptions stores all option input from `nerdctl compose stop`
+// Options stores all option input from `nerdctl compose stop`
```
Comment (djdongjin) [inline pkg/composer/stop.go:32]: change back

File: pkg/logging/log_viewer.go [hunk 17]
```diff
@@ -28,7 +28,9 @@ import (
 )
 
 // Type alias for functions which write out logs to the provided stdout/stderr Writers.
-// Depending on the provided `LogViewOptions.Follow` option, the function may block
+//
+//	# Options.Follow` option, the function may block
+//
```
Comment (djdongjin) [inline pkg/logging/log_viewer.go:33]: change back

File: pkg/logging/log_viewer.go [hunk 18]
```diff
@@ -101,7 +103,8 @@ func (lvo *LogViewOptions) Validate() error {
 }
 
 // Implements functionality for loading the logging configuration and
-// fetching/outputting container logs based on its internal LogViewOptions.
+//
+//	Options.
```
Comment (djdongjin) [inline pkg/logging/log_viewer.go:107]: change back

File: pkg/logging/log_viewer.go [hunk 19]
```diff
@@ -113,7 +116,8 @@ type ContainerLogViewer struct {
 	stopChannel chan os.Signal
 }
 
-// Validates the given LogViewOptions, loads the logging config for the
+//	Options, loads the logging config for the
+//
```
Comment (djdongjin) [inline pkg/logging/log_viewer.go:120]: change back

File: pkg/mountutil/mountutil_freebsd.go [hunk 20]
```diff
@@ -34,7 +34,8 @@ func getUnprivilegedMountFlags(path string) ([]string, error) {
 // FreeBSD doesn't support bind mounts.
 const DefaultPropagationMode = ""
 
-// parseVolumeOptions parses specified optsRaw with using information of
+//	Options parses specified optsRaw with using information of
+//
```
Comment (djdongjin) [inline pkg/mountutil/mountutil_freebsd.go:38]: change back

File: pkg/mountutil/mountutil_linux.go [hunk 21]
```diff
@@ -82,7 +82,8 @@ func getUnprivilegedMountFlags(path string) ([]string, error) {
 // See also: https://github.com/moby/moby/blob/v20.10.7/volume/mounts/linux_parser.go#L145
 const DefaultPropagationMode = "rprivate"
 
-// parseVolumeOptions parses specified optsRaw with using information of
+//	Options parses specified optsRaw with using information of
+//
```
Comment (djdongjin) [inline pkg/mountutil/mountutil_linux.go:86]: change back

File: pkg/mountutil/mountutil_linux.go [hunk 22]
```diff
@@ -97,8 +98,8 @@ func getMountInfo(dir string) (mount.Info, error) {
 	return mount.Lookup(sourcePath)
 }
 
-// parseVolumeOptionsWithMountInfo is the testable implementation
-// of parseVolumeOptions.
+// OptionsWithMountInfo is the testable implementation
+// Options.
```
Comment (djdongjin) [inline pkg/mountutil/mountutil_linux.go:102]: change back

File: pkg/mountutil/mountutil_linux_test.go [hunk 23]
```diff
@@ -28,7 +28,7 @@ import (
 	is "gotest.tools/v3/assert/cmp"
 )
 
-// TestParseVolumeOptions tests volume options are parsed as expected.
+// Options tests volume options are parsed as expected.
```
Comment (djdongjin) [inline pkg/mountutil/mountutil_linux_test.go:31]: change back

**Review by djdongjin — CHANGES_REQUESTED** [review]
Changes in `pkg/*` packages (except `pkg/api` and `pkg/cmd`) seems unexpected. Can you revert these changes?

File: pkg/composer/port.go [hunk 24]
```diff
@@ -25,6 +25,7 @@ import (
 )
 
 // PortOptions has args for getting the public port of a given private port/protocol
+//
```
Comment (djdongjin) [inline pkg/composer/port.go:28]: can you also remove this change? 😅

File: pkg/composer/rm.go [hunk 25]
```diff
@@ -31,6 +31,7 @@ import (
 )
 
 // RemoveOptions stores all options when removing compose containers:
+//
```
Comment (djdongjin) [inline pkg/composer/rm.go:34]: and this one

**Review by djdongjin — APPROVED** [review]
LGTM aside from 2 minor changes, thanks

File: cmd/nerdctl/build.go [hunk 26]
```diff
@@ -68,91 +68,91 @@ If Dockerfile is not present and -f is not specified, it will look for Container
 	return buildCommand
 }
 
-func processBuildCommandFlag(cmd *cobra.Command, args []string) (*types.BuildCommandOptions, error) {
+func processBuildCommandFlag(cmd *cobra.Command, args []string) (types.BuildOptions, error) {
 	globalOptions, err := processRootCmdFlags(cmd)
 	if err != nil {
-		return nil, err
+		return types.BuildOptions{}, err
```
Comment (AkihiroSuda) [inline cmd/nerdctl/build.go:74]: `BuilderBuildOptions` for consistency with other commands

File: cmd/nerdctl/run.go [hunk 27]
```diff
@@ -741,23 +741,25 @@ func createContainer(ctx context.Context, cmd *cobra.Command, client *containerd
 
 // When refactor `nerdctl run`, this func should be removed and replaced by
 // creating a `PullCommandOptions` directly from `RunCommandOptions`.
-func processPullCommandFlagsInRun(cmd *cobra.Command) (types.PullCommandOptions, error) {
+func processPullCommandFlagsInRun(cmd *cobra.Command) (types.PullOptions, error) {
```
Comment (AkihiroSuda) [inline cmd/nerdctl/run.go:744]:
```suggestion
func processPullCommandFlagsInRun(cmd *cobra.Command) (types.ImagePullOptions, error) {
```

File: pkg/api/types/build_types.go [hunk 28]
```diff
@@ -16,8 +16,13 @@
 
 package types
 
-// BuildCommandOptions specifies options for `nerdctl (image/builder) build`.
-type BuildCommandOptions struct {
+import "io"
+
+// BuildOptions specifies options for `nerdctl (image/builder) build`.
+type BuildOptions struct {
```
Comment (AkihiroSuda) [inline pkg/api/types/build_types.go:22]:
```suggestion
type BuilderBuildOptions struct {
```

Probably the file name should be `builder_types.go`

File: pkg/api/types/info_types.go [hunk 29]
```diff
@@ -16,8 +16,12 @@
 
 package types
 
-// InfoCommandOptions specifies options for `nerdctl (system) info`.
-type InfoCommandOptions struct {
+import "io"
+
+// InfoOptions specifies options for `nerdctl (system) info`.
+type InfoOptions struct {
```
Comment (AkihiroSuda) [inline pkg/api/types/info_types.go:22]:
```suggestion
type SystemInfoOptions struct {
```
(`system_types.go`)

File: pkg/api/types/load_types.go [hunk 30]
```diff
@@ -16,8 +16,12 @@
 
 package types
 
-// LoadCommandOptions specifies options for `nerdctl (image) load`.
-type LoadCommandOptions struct {
+import "io"
+
+// LoadOptions specifies options for `nerdctl (image) load`.
+type LoadOptions struct {
```
Comment (AkihiroSuda) [inline pkg/api/types/load_types.go:22]:
```suggestion
type ImageLoadOptions struct {
```
(`image_types.go`)

File: pkg/api/types/pull_types.go [hunk 31]
```diff
@@ -16,8 +16,12 @@
 
 package types
 
-// PullCommandOptions specifies options for `nerdctl (image) pull`.
-type PullCommandOptions struct {
+import "io"
+
+// PullOptions specifies options for `nerdctl (image) pull`.
+type PullOptions struct {
```
Comment (AkihiroSuda) [inline pkg/api/types/pull_types.go:22]:
```suggestion
type ImagePullOptions struct {
```
(`image_types.go`)

File: pkg/cmd/builder/build.go [hunk 32]
```diff
@@ -62,10 +62,10 @@ func Build(ctx context.Context, options *types.BuildCommandOptions, stdin io.Rea
 			return err
 		}
 	} else {
-		buildctlCmd.Stdout = stdout
+		buildctlCmd.Stdout = options.Stdout
```
Comment (AkihiroSuda) [inline pkg/cmd/builder/build.go:65]: Should be set to `os.Stdout` if nil

Comment (Zheaoli) [reply pkg/cmd/builder/build.go:65]: I think this is not related to this PR, I can make a split PR later

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**